wagtail/admin/views/reports/page_types_usage.py

```diff
@@ -0,0 +1,191 @@
+    ).order_by("-timestamp", "-pk")
+
+    queryset = queryset.annotate(
+        count=Count("pages", filter=Q(pages__locale__language_code=language_code)),
+        last_edited_page=Subquery(latest_edit_log.values("page")[:1]),
+        last_edited_by=Subquery(latest_edit_log.values("user")[:1]),
+    )
+
+    return queryset
+
+
+class LocaleFilter(django_filters.ChoiceFilter):
+    def filter(self, qs, value):
+        # The filtering is handled in the filter_queryset method
+        return qs
+
+
+class SiteFilter(django_filters.ChoiceFilter):
+    def filter(self, qs, value):
+        # The filtering is handled in the filter_queryset method
+        return qs
+
+
+class PageTypesUsageReportFilterSet(WagtailFilterSet):
+    page_locale = LocaleFilter(
+        label=_("Locale"),
+        choices=_get_locale_choices,
+        empty_label=None,
+        null_label=_("All"),
+        null_value="all",
+    )
+    site = SiteFilter(
+        label=_("Site"),
+        choices=_get_site_choices,
+        empty_label=None,
+        null_label=_("All"),
+        null_value="all",
+    )
+
+    class Meta:
+        model = ContentType
+        fields = ["page_locale", "site"]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.sites = {
+            site.pk: site for site in Site.objects.all().prefetch_related("root_page")
+        }
+        self.sites_filter_enabled = True
+        if len(self.sites) == 1:
+            # If there is only one site, we don't need to show the site filter
+            self.sites_filter_enabled = False
+            del self.filters["site"]
+
+    def filter_queryset(self, queryset):
+        queryset = super().filter_queryset(queryset)
+
+        if self.sites_filter_enabled:
+            site = self.form.cleaned_data["site"]
+
+            if site and site != self.filters["site"].null_value:
+                site_obj = self.sites[int(site)]
+                queryset = queryset.filter(
+                    pages__path__startswith=site_obj.root_page.path
+                )
+
+        locale = self.form.cleaned_data["page_locale"]
+
+        if locale and locale != self.filters["page_locale"].null_value:
+            queryset = _annotate_last_edit_info_by_locale(queryset, locale)
+        else:
+            queryset = _annotate_last_edit_info(queryset)
+
+        return queryset
+
+
+class PageTypesUsageReportView(ReportView):
+    template_name = "wagtailadmin/reports/page_types_usage.html"
+    title = _("Page types usage")
+    header_icon = "doc-empty-inverse"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.user_model = get_user_model()
+        self.page_models = [model.__name__.lower() for model in get_page_models()]
+        self.i18n_enabled = getattr(settings, "WAGTAIL_I18N_ENABLED", False)
+
+    def get_filterset_class(self):
+        if self.i18n_enabled:
```

Comment:
Now that we have a `site` filter, wouldn't we want to return the class here regardless?

Reply:
Yes, thanks. Nice catch. I'll put this check into the filterset itself.

Reply:
Done!